pkg/crypto/tinkcrypto/primitive/composite/composite_common.go

```diff
@@ -39,3 +47,29 @@ type PublicKey struct {
 	Curve string `json:"curve,omitempty"`
 	Type  string `json:"type,omitempty"`
 }
+
+// GetCurveType is a utility function that converts a string EC curve name into an EC curve proto type
+func GetCurveType(curve string) (commonpb.EllipticCurveType, error) {
+	switch curve {
+	case "secp256r1", "NIST_P256", "P-256", "EllipticCurveType_NIST_P256":
+		return commonpb.EllipticCurveType_NIST_P256, nil
+	case "secp384r1", "NIST_P384", "P-384", "EllipticCurveType_NIST_P384":
+		return commonpb.EllipticCurveType_NIST_P384, nil
+	case "secp521r1", "NIST_P521", "P-521", "EllipticCurveType_NIST_P521":
+		return commonpb.EllipticCurveType_NIST_P521, nil
+	default:
+		return 0, fmt.Errorf("curve %s not supported", curve)
```

Comment:
Use `EllipticCurveType_UNKNOWN_CURVE` instead of `0`. Also, I would remove the error since we have `EllipticCurveType_UNKNOWN_CURVE`